### DISCUSSION TITLE
automatic tap regulation disabled

### DISCUSSION THREAD

**Discussion by willemijnbrus** [PowerGridModel/.github#38] [Q&A] [answered]
hi pgm team,
We added an autmatic tap regulator like in the example notebook (see code below). 
we want to make it a choice whether the model runs with the given information about the tap position (transformer["tap_pos"] ) or with automatic tap regulation. Is it correct that when we put TapChangingStrategy.disabled in the calculate_power_flow() function, the loadflow will use the tap pos assigned to the transformer and ignore the information we added on the transformer_tap_regulator?

    transformer_tap_regulator = initialize_array(
        DatasetType.input, ComponentType.transformer_tap_regulator, 1
    )
    transformer_tap_regulator["id"] = [888888888]
    transformer_tap_regulator["regulated_object"] = transformer_id
    transformer_tap_regulator["status"] = [1]
    transformer_tap_regulator["control_side"] = [BranchSide.to_side]
    transformer_tap_regulator["u_set"] = [400.0]
    transformer_tap_regulator["u_band"] = [20.0]
    transformer_tap_regulator["line_drop_compensation_r"] = [0.0]
    transformer_tap_regulator["line_drop_compensation_x"] = [0.0]

**Comment by mgovers** [comment, marked as answer]
Hi @willemijnbrus ,

Indeed, the [`TapChangingStrategy.disabled`](https://power-grid-model.readthedocs.io/en/stable/api_reference/python-api-reference.html#power_grid_model.enum.TapChangingStrategy.disabled) enum value has exactly that purpose. In fact, the [`PowerGridModel.calculate_power_flow()`](https://power-grid-model.readthedocs.io/en/stable/api_reference/python-api-reference.html#power_grid_model.PowerGridModel.calculate_power_flow) function by default uses that value as well (see image below).

I hope this answers your question.

![image](https://github.com/user-attachments/assets/629f780e-9068-4243-9fd8-c30c01c9d635)

**Reply by willemijnbrus** [reply]
Thank you Martijn!
